packages/labs/motion/src/view-transition-update.ts

```diff
@@ -0,0 +1,48 @@
+import {LitElement} from 'lit';
+
+type Constructor<T> = new (...args: any[]) => T;
+
+/** Extend a LitElement to use view transitions (if available) on DOM updates */
+export const ViewTransitionUpdate =
+    <T extends Constructor<LitElement>>(superClass: T) => {
+    class LitElementTransition extends superClass {
+        private lifecycleRenderTransition: ViewTransition;
+  
+        async performUpdate() {
+            if (!(document as DocumentWithTransition).startViewTransition)
+                return await super.performUpdate();
+    
+            if (this.lifecycleRenderTransition)
+                await this.lifecycleRenderTransition.finished;
```

Comment:
Given the discussion thread, we'll need a try/catch and a comment on why this is the right promise to wait instead of say updateCallbackDone.

And it is a good opinion to await finished? It makes sense to me, but maybe we need an option? As this is labs we could go with this and wait for feedback.

Reply:
try-catch and comment added in a1b5fa32a4916a1b704b9289755b7e2f891bdc76

I'm not certain about `await finished` vs `await updateCallbackDone` vs `skipTransition()` either.